### ISSUE TITLE
getNeighbors Count method

### ISSUE THREAD

**Issue by Chris-Cooney** [aestears/plantTracker#7]
I am trying to use the getNeighbors function in PlantTracker, the data works when calculating the area occupied by species within the buffer area, but does not work when method = "count". 

Any advice or help would be amazing, thank you!!

![Screen Shot 2022-10-06 at 10 36 56 am](https://user-images.githubusercontent.com/92921661/194183077-1a149c9a-eb42-4ca6-b8ec-6929f9589240.png)

**Comment by aestears** [comment]
Hi Chris, thanks for getting in touch! Could you include a small subset of the data you're using in the function? (it looks like it's the "taglab" data frame). Without a reproducible example, it's difficult to diagnose the problem.

**Comment by Chris-Cooney** [comment]
Hi Alice,

I've attached the first 5 rows of the data as an example, for context as this may change how the function works. There is currently only 1 site and year for the data and "TL_Genet" is the unique numerical id for each polygon.

Thank you!!

[Taglab example.zip](https://github.com/aestears/plantTracker/files/9781640/Taglab.example.zip)

**Comment by aestears** [comment]
Hi Chris, 
I was able to get the getNeighbors() function to run with the first five rows of data, so I'm not too sure what the issue is. Can you run the function with those five rows only, or does the problem only arise when you use the entire dataset? 

`library(devtools)
devtools::install_github("aestears/plantTracker")

library(plantTracker)
library(sf)

dat <- st_read(dsn = "./Downloads/Taglab example/", layer = "example") 

neDat <- getNeighbors(dat = dat, buff = .5, method = "count", compType = "allSpp", output = "bySpecies", trackID = "TL_Genet", species = "Species", quad = "Quad", year = "Year", site = "Site", geometry = "geometry")`

**Comment by aestears** [comment]
Hi Chris, 
While I couldn't replicate the error you are describing with just the first five rows of the dataset, I think I may have identified the issue. The function was having trouble counting neighbors when there was just one individual of a species in the quadrat in a given year. I've fixed that issue in a new release, let me know if the function works after re-downloading the package!

**Comment by Chris-Cooney** [comment]
HI Alice,

Yes, you're correct it does work. It still doesn't seem to like the larger data set. When I searched through the data created by the get neighbours count method, I find that in the "overlapSpp_list" it appears only to count the first 3 of 9 species.

I've attached the full data set so that hopefully the issue is replicated and can be solved.

[taglabTest.zip](https://github.com/aestears/plantTracker/files/9799700/taglabTest.zip)

Thank you,
Chris

**Comment by aestears** [comment]
Hi Chris, 
I fixed another bug (that was related to the first issue I described, but arose in a slightly different context), and the function now runs with the entire dataset you shared. I've included a built version of the package with this update here, does it fix the issue for you also? I haven't yet updated the package on GitHub, but I will if the fix works for you. 

As far as the issue you described above "I find that in the "overlapSpp_list" it appears only to count the first 3 of 9 species", I'm not sure what the issue is there, and I wasn't able to replicate it with the data you sent me. With this updated version of the function counts were generated for all species. Let me know if the issue persists for you. 

-Alice 
[plantTracker_1.0.1.tar.gz](https://github.com/aestears/plantTracker/files/9823285/plantTracker_1.0.1.tar.gz)
